### DISCUSSION TITLE
API to dump config object as .ini

### DISCUSSION THREAD

**Discussion by abitrolly** [sqlalchemy/alembic#933] [Ideas]
Config object in API doesn't allow to dump its contents for debugging. At least the doc is silent about it. The ideal way would be to dump it in .ini format for comparison.

**Comment by zzzeek** [comment]
hi -

Alembic's config as mentioned previously is a thin facade over the [Python std lib configparser object](https://docs.python.org/3/library/configparser.html).   It's accessible by the publicly documented [file_config](https://alembic.sqlalchemy.org/en/latest/api/config.html?highlight=config%20file_config#alembic.config.Config.file_config) attribute.  When you have the ConfigParser you can use the [write](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.write) method:


```py

with open('example.ini', 'w') as configfile:
    alembic_config.file_config.write(configfile)
```

hope this helps

**Reply by abitrolly** [reply]
So why not to expose API that allows to dump the config to stdout?

**Reply by zzzeek** [reply]
it's redundant, unnecessary, and out of scope for the project.  

I understand you are trying to achieve something with the warehouse project but you aren't being held back from any of these things by alembic - alembic is transparent.  I would ask that you please work with the API as it stands now as we are an independent, non-commercial project with limited resources.  thanks!
